### PR TITLE
Initial version config file support

### PR DESCRIPTION
**Pull request by balloob**
This is an initial version to add support for configuring Farcy via configuration files. It supports a structure similar to [praw](http://praw.readthedocs.org/en/latest/pages/configuration_files.html): a default configuration section and a section per repository.

 - Configuration priority is: command line arguments > repo specific section > default section. They overwrite one another, they will not be merged.
 - It will look for a configuration file at `~/.config/farcy/farcy.conf`.
 - The default section supports 1 extra parameter: `repository`. This will be used if no repository given as command line argument.
 - Debug expects a true-ish value: `true` or `1`

Example:

```conf
[default]
repository: appfolio/farcy
limit_users: balloob
exclude_paths: node_modules, vendor, db

[appfolio/farcy]
limit_users: balloob, bboe
start_event: 100
debug: 1
exclude_paths: node_modules, some_other_path
limit_users: balloob, bboe
log_level: DEBUG
pr_issue_report_limit: 20
```

### CODE REVIEW

File: farcy/__init__.py [hunk 1]
```diff
@@ -402,12 +402,63 @@ def run(self):
 def main():
     """Provide an entry point into Farcy."""
     args = docopt(__doc__, version=VERSION_STR)
-    limit_users = process_user_list(args['--limit-user'])
+
+    not_set_value = {
```
Comment (bboe) [inline farcy/__init__.py]: Can we move all of this logic into a configuration class?

**Comment by bboe** [comment]
This addition looks awesome. I hope we can move the configuration parsing logic into it's own class or at least a method, and then add some tests with respect to enforcing priority between items as well as the possible error cases.

File: farcy/__init__.py (hunk 1 above)
Comment (balloob) [inline farcy/__init__.py]: Will do.

**Comment by balloob** [comment]
Extracted the configuration into a separate class in `helpers.py` and added test coverage.

**Pull request merged**